controllers/machinehealthcheck_controller.go

```diff
@@ -299,10 +299,12 @@ func (r *MachineHealthCheckReconciler) PatchHealthyTargets(ctx context.Context,
 			// Get remediation request object
 			obj, err := r.getExternalRemediationRequest(ctx, m, t.Machine.Name)
 			if err != nil {
-				if apierrors.IsNotFound(errors.Cause(err)) {
-					continue
+				if !apierrors.IsNotFound(errors.Cause(err)) {
+					wrappedErr := errors.Wrapf(err, "failed to fetch remediation request for machine %q in namespace %q within cluster %q", t.Machine.Name, t.Machine.Namespace, t.Machine.ClusterName)
+					logger.Error(wrappedErr, "")
+					errList = append(errList, wrappedErr)
```

Comment:
Do we need to log this error out?

Reply:
need

Reply:
I don't think we necessarily need to log it out here, given that's going to be logged in the controller already. The best way to surface these errors would be in an condition though.

Reply:
@vincepri removed

Reply:
/vincepri